namer/namer.cc

```diff
@@ -118,6 +118,7 @@ class NameInserter {
         }
         if (parsedArg.block) {
             argInfo.flags.isBlock = true;
+            localExpr = make_unique<ast::BlockArg>(parsedArg.loc, move(localExpr));
```

Comment:
This was wrong before and was causing problems in the namer test, because after we ran the namer we would have modified the tree in-place in such a way that it turned a block arg into a positional arg.

Reply:
can we add an isolated test for this(and, in best case extract it to separate PR)